firebase/ManageEvents.ts

```diff
@@ -31,6 +30,7 @@ const formatEvent = (doc): Event => {
 
   const event = {
     ...data2,
+    uid: doc.id,
```

Comment:
this should be generated directly and we should not allow it to be passed as an argument. It's essential to ensure uniqueness of the uid